src/commands/copyRelativePathToClipboard.ts

```diff
@@ -0,0 +1,26 @@
+import { env, TextEditor, Uri } from 'vscode';
+import { Commands } from '../constants';
+import type { Container } from '../container';
+import { command } from '../system/command';
+import { ActiveEditorCommand, getCommandUri } from './base';
+
+@command()
+export class CopyRelativePathToClipboardCommand extends ActiveEditorCommand {
+	constructor(private readonly container: Container) {
+		super(Commands.CopyRelativePathToClipboard);
+	}
+
+	async execute(editor?: TextEditor, uri?: Uri) {
+		uri = getCommandUri(uri, editor);
+		const repoPath = this.container.git.getBestRepository(editor)?.path;
+		let relativePath = '';
+		if (uri != null && repoPath != null) {
```

Comment:
Instead of messing with the path here, you should be able to call `this.container.git.getRelativePath(uri, repoPath)`